### PR TITLE
Fix contradictory advice in "detach" docs; cut redundancy in "attach"

### LINKED ISSUE #115525: Conflicting API docs for RenderObjects
The following screenshot shows the current API docs for the `detach()` method in `AbstractNode`. The instructions appear to conflict. We're told "first call their inherited detach method, and then detach all their children", but we're also told "this method should end with a call to the inherited method, as in super.detach()". We're told that "super.detach()" should be the first thing called, but also the last thing called.

There may be similar issues with other docs. The line "Implementations of this method should end with a call to the inherited method, as in super.detach()." is repeated in a similar fashion for some other similar APIs. I suggest doing an audit.

Screen Shot 2022-11-16 at 8 14 02 PM

### PR DESCRIPTION
**Pull request by gnprice**
Fixes #115525.

On [AbstractNode.detach] and its two progeny [RenderNode.detach]
and [Layer.detach], the docs said both to call the inherited method
before detaching children, and to end by doing so.  The former
advice is what's enforced by an assertion in the base implementation,
so cut out the other.

The corresponding [attach] methods redundantly said twice to
call the inherited method first, so cut the redundancy.
Leave in place the version more recently added (in #76021), because
that PR shows the old version must have been easy to overlook.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Hixie** [comment]
Would it make sense to phrase it something like "Implementations in subclasses should attach their children after calling the superclass method" (as opposed to "must start with calling the superclass method")? (Maybe with better phrasing, that's not my best work!) The point being to avoid implying that doing unrelated work before calling the superclass is a problem.

**Comment by Hixie** [comment]
SGTM.

**Comment by gnprice** [comment]
Cool. Pushed a revision with that text.

**Review by goderbauer — APPROVED** [review]
LGTM
